src/lib/Parser.hs

```diff
@@ -285,6 +337,40 @@ decl = do
   rhs <- sym "=" >> blockOrExpr
   return $ lhs rhs
 
+interfaceInstance :: Parser UDecl
+interfaceInstance = do
+  keyWord InstanceKW
+  (lhs, pos) <- withPos simpleLet
+  keyWord WhereKW
+  rhs <- withSrc $ URecord <$> interfaceRecordFields "="
+  let ULet _ (_, Just (WithSrc _  ann)) _ = lhs rhs
+  let ULet _ (p, ann') rhs' = lhs $ mkRhs ann rhs
+  let (ann'', rhs'') = addImplicitImplicitArgs pos ann' rhs'
+  return $ ULet InstanceLet (p, ann'') rhs''
+  where
+    mkRhs (UPi _ _ (WithSrc _ lrhs)) rhs = mkRhs lrhs rhs
+    mkRhs (UApp _ (WithSrc _ (UVar v)) _) rhs =
+      (var . nameToStr . mkInterfaceConsName . varName) v `mkApp` rhs
```

Comment:
I'm not sure if I follow this code. Can you please explain/add a comment? Why are you traversing the type?

Reply:
Hopefully addressed in `4a8b9d4`!